apps/roboshield/payload.config.ts

```diff
@@ -123,23 +101,52 @@ export default buildConfig({
         },
       ],
       uploadsCollection: "media",
-      generateTitle: ({ doc }: any) => doc?.title?.value as string,
+      generateTitle: ({ doc }: any) => (doc?.title as string) || "",
       generateURL: ({ doc }: any) =>
-        doc?.slug?.value ? `${appURL}/${doc.slug.value}` : undefined,
-    } as any),
-    nestedDocs({
+        doc?.slug ? `${appURL}/${doc?.slug}` : "",
+    }),
+    nestedDocsPlugin({
       collections: ["pages"],
-      generateLabel: (_, doc) => doc.title as string,
+      generateLabel: (_, doc) => (doc?.title as string) || "",
       generateURL: (docs) =>
-        docs.reduce((url, doc) => `${url}/${doc.slug}`, ""),
+        docs.reduce((url, doc) => `${url}/${doc?.slug}`, ""),
     }),
-  ] as any[],
+    s3Storage({
+      collections: {
+        media: true,
+        "media-with-prefix": {
+          prefix: "media",
+        },
+      },
+      bucket: process.env.S3_BUCKET ?? "",
+      config: {
+        credentials: {
+          accessKeyId: process.env.S3_ACCESS_KEY_ID ?? "",
+          secretAccessKey: process.env.S3_SECRET_ACCESS_KEY ?? "",
+        },
+        region: process.env.S3_REGION ?? "",
+      },
+    }),
+  ],
+  ...(smtpAuthPass
+    ? {
+        email: nodemailerAdapter({
+          defaultFromAddress: smtpFromAddress,
+          defaultFromName: smtpFromName,
+          transportOptions: {
+            host: smtpHost,
+            port: smtpPort,
+            auth: {
+              user: smtpAuthUser,
+              pass: smtpAuthPass,
+            },
+          },
+        }),
+      }
+    : {}),
+  secret: process.env.PAYLOAD_SECRET || "",
```

Comment:
```suggestion
  secret: process.env.PAYLOAD_SECRET,
```

It's best for the application fails if secret is not set than to start with an empty secret.

Reply:
I think there isn't much we can do here, `secret` only supports `string` type and is required